core/test_utils.py

```diff
@@ -70,3 +71,12 @@ def test_constant_step_size_zero_step_size(self):
         new_direction = utils.constant_step_size(np.array([1, 1]), 0)
         expected_direction = np.array([0, 0])
         np.testing.assert_almost_equal(new_direction, expected_direction)
+
+    @mock.patch("core.utils.model_interface.Model", autospec=True)
+    def test_random_poi(self, mock_model):
+        # Test that random_poi selects the negatively classified point
+        dataset = pd.DataFrame({"a": [0, 1], "label": [-1, 1]})
+        mock_model.predict.return_value = np.array([1, -1])
```

Comment:
This placeholder `dataset` was ambiguous to me in another recent PR. It may be worth adding a comment above that `dataset` "a" is a dataset of two 1D points (looking at the label reveals this, but there also isn't a comment in this region of the code base alluding to the binary classification assumption).